components/match2/commons/match_group_input.lua

```diff
@@ -724,4 +727,129 @@ function MatchGroupInput._getCasterInformation(name, flag, displayName)
 	}
 end
 
+-- Parse map veto input
+---@param match table
+---@param allowedVetoes string[]?
+---@return {type:string, team1: string?, team2:string?, decider:string?, vetostart:string?}[]?
+function MatchGroupInput.getMapVeto(match, allowedVetoes)
+	if not match.mapveto then return nil end
+
+	allowedVetoes = allowedVetoes or DEFAULT_ALLOWED_VETOES
+
+	match.mapveto = Json.parseIfString(match.mapveto)
+
+	local vetoTypes = mw.text.split(match.mapveto.types or '', ',')
+	local deciders = mw.text.split(match.mapveto.decider or '', ',')
+	local vetoStart = match.mapveto.firstpick or ''
+	local deciderIndex = 1
+
+	local data = {}
+	for index, vetoType in ipairs(vetoTypes) do
+		vetoType = mw.text.trim(vetoType):lower()
+		if not Table.includes(allowedVetoes, vetoType) then
+			return nil -- Any invalid input will not store (ie hide) all vetoes.
+		end
+		if vetoType == 'decider' then
+			table.insert(data, {type = vetoType, decider = deciders[deciderIndex]})
+			deciderIndex = deciderIndex + 1
+		else
+			table.insert(data, {type = vetoType, team1 = match.mapveto['t1map'..index], team2 = match.mapveto['t2map'..index]})
```

Comment:
Should we make this for generic opponents? (Would be good to adjust keys as well, either t->o or have aliases)
```suggestion
			table.insert(data, {type = vetoType, opponent1 = match.mapveto['t1map'..index], opponent2 = match.mapveto['t2map'..index]})
```

Reply:
1. would change storage, i.e. not nice for api consumers
2. not sure where this data is used, in matchsummary for sure (relatively easy to adjust when switching to using this function in input processing) but could have usage outside and shit to check/search for it

Reply:
IMO the commons part should still be as reusable as possible, so should allow/respect non-team opponent types.
we could add the corresponding input?
Something like this
```suggestion
			table.insert(data, {type = vetoType, team1 = match.mapveto['t1map'..index], p1 = match.mapveto['p1map'..index], team2 = match.mapveto['t2map'..index], p2 = match.mapveto['p1map'..index]})
```

Reply:
i would be okay with this (modulo line break)

@Rathoz opinion?

Reply:
For now, keep the same naming as we already have, both for input and storage. It's still usable in 1v1s, even if the parameter name is strange.

Later on we can transition first the wiki input, and later the db storage. But at that point just might make move it out of extradata and put in its own field

Reply:
The second option would leave current naming untouched, but would be less confusing for input, what speaks against doing so?

Reply:
Not a fan of storing it under 2 different names

Reply:
> Not a fan of storing it under 2 different names

It would be only one or the other, but not both.
I'm also more after the change on input parameters to streamline it for contributors, less about the storage

Reply:
imo OOS for now
we can look into supporting that in a follow up
mind several wikis use other veto systems anyways

Reply:
agreed, not a bad idea to change it, but OOS for now, this PR is just extracting

Reply:
opened #4257 to keep track of it